src/static/helpers/slas.ts

```diff
@@ -0,0 +1,409 @@
+    `${slasClient.clientConfig.parameters.clientId}:${credentials.clientSecret}`
+  )}`;
+
+  const options = {
+    headers: {
+      Authorization: authorization,
+    },
+    body: {
+      grant_type: "client_credentials",
+      ...(usid && { usid: usid }),
+    },
+  };
+
+  return slasClient.getAccessToken(options);
+}
+
+/**
+ * A single function to execute the ShopperLogin Public Client Guest Login with proof key for code exchange flow as described in the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/guide/slas-public-client.html).
+ * @param slasClient a configured instance of the ShopperLogin SDK client.
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.redirectURI - Per OAuth standard, a valid app route. Must be listed in your SLAS configuration. On server, this will not be actually called
+ * @param parameters.usid? - Unique Shopper Identifier to enable personalization.
+ * @returns TokenResponse
+ */
+export async function loginGuestUser(
+  slasClient: ISlasClient,
+  parameters: {
+    redirectURI: string;
+    usid?: string;
+  }
+): Promise<TokenResponse> {
+  const codeVerifier = createCodeVerifier();
+
+  const authResponse = await authorize(slasClient, codeVerifier, {
+    redirectURI: parameters.redirectURI,
+    hint: "guest",
+    ...(parameters.usid && { usid: parameters.usid }),
+  });
+
+  const tokenBody: TokenRequest = {
+    client_id: slasClient.clientConfig.parameters.clientId,
+    code: authResponse.code,
+    code_verifier: codeVerifier,
+    grant_type: "authorization_code_pkce",
+    redirect_uri: parameters.redirectURI,
+    usid: authResponse.usid,
+  };
+
+  return slasClient.getAccessToken({ body: tokenBody });
+}
+
+/**
+ * A single function to execute the ShopperLogin Private Client Registered User B2C Login as described in the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/guide/slas-private-client.html).
+ * @param slasClient - a configured instance of the ShopperLogin SDK client.
+ * @param credentials - the shopper username and password for login and client secret for additional authentication
+ * @param credentials.username - the id of the user to login with
+ * @param credentials.password - the password of the user to login with
+ * @param credentials.clientSecret - secret associated with client ID
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.redirectURI - Per OAuth standard, a valid app route. Must be listed in your SLAS configuration. On server, this will not be actually called
+ * @param parameters.usid? - optional Unique Shopper Identifier to enable personalization
+ * @returns TokenResponse
+ */
+export async function loginRegisteredUserB2Cprivate(
+  slasClient: ISlasClient,
+  credentials: {
+    username: string;
+    password: string;
+    clientSecret: string;
+  },
+  parameters: {
+    redirectURI: string;
+    usid?: string;
+  }
```

Comment:
Not 100% on this, but iirc defining object types within function signatures like this can cause issues with the generated docs. But it may have been _nested_ objects (e.g. `options: { credentials: { ... }, parameters: { ... } }`), and it's been more than a year, so maybe the bug's been patched.